File: the_descent/descent.java
package the_descent;
import java.util.*;

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/
class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        ArrayList<Integer> hauteurs = new ArrayList<>(); // Initialisez la liste

        // game loop
        while (true) {
            hauteurs.clear(); // Videz la liste à chaque itération
            for (int i = 0; i < 8; i++) {
                int mountainH = in.nextInt(); // represents the height of one mountain.
                hauteurs.add(mountainH);
            }

            // Trouver la montagne la plus haute
            int max_index = 0;
            for (int i = 0; i < hauteurs.size(); i++) {
                if (hauteurs.get(i) > hauteurs.get(max_index)) {
                    max_index = i;
                }
            }

            // Imprimer l'index de la montagne la plus haute
            System.out.println(max_index);
        }
    }
}

File: the_descent/descent.py
import sys
import math

# The while loop represents the game.
# Each iteration represents a turn of the game
# where you are given inputs (the heights of the mountains)
# and where you have to print an output (the index of the mountain to fire on)
# The inputs you are given are automatically updated according to your last actions.


# game loop
while True:
    max_height = 0  
    hauteur_list = []
    for i in range(8):
        mountain_h = int(input())  # represents the height of one mountain.
        hauteur_list.append(mountain_h)
        if mountain_h > max_height:  # vérifier si la hauteur actuelle est supérieure à la hauteur maximale
            max_height = mountain_h  # mettre à jour la hauteur maximale si nécessaire

    # Write an action using print
    # To debug: print("Debug messages...", file=sys.stderr, flush=True)

    # The index of the mountain to fire on.
    res = hauteur_list.index(max_height)  # obtenir l'indice de la hauteur maximale
    print(res)  # imprimer l'indice de la hauteur maximale
